protocol/x/clob/keeper/liquidations.go

```diff
@@ -152,6 +155,27 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 			)
 			return err
 		}
+		clobPairId := k.mustGetClobPairForPerpetualId(ctx, perpetualId).Id
+		subticks := liquidationOrder.GetOrderSubticks()
+		isBuy := liquidationOrder.IsBuy()
+		// Send on-chain deleveraging event for each fill.
+		for _, fill := range fills {
+			k.GetIndexerEventManager().AddTxnEvent(
```

Comment:
@jayy04 Is there any preference on whether to emit the event here as compared to where the CometBFT events are emitted ([here](https://github.com/dydxprotocol/v4-chain/blob/f892a43fccdad4bcdf4e1b79fc122680bbc888ab/protocol/x/clob/keeper/deleveraging.go#L436))?

Reply:
wait just to make sure we are on the same page - this is happening in `PrepareCheckState` / CheckState, so deleveraging fills are not on-chain yet. I don't think we should be sending an on-chain event?

deleveraging fills, similar to other order matches, get persisted to consensus state in DeliverTx [here](https://github.com/dydxprotocol/v4-chain/blob/main/protocol/x/clob/keeper/process_operations.go#L619)

Reply:
and for my understanding - CometBFT events are offchain events?

Reply:
CometBFT events are not stored in consensus so yes they would be off-chain.
The indexer events are added to a `TransientStore` and only sent in the EndBlocker call back.